### PR TITLE
doc: update readme and pre post install

### PR DESCRIPTION
**Pull request by smomin**
- update readme with clear param descriptions
- sync pre/post docs with changes from readme
- sync extension.yaml with changes from readme

### CODE REVIEW

File: POSTINSTALL.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ This extension only sends the documents that have been changed -- it does not in
 
 The import script can read all existing documents in a Cloud Firestore collection and insert them into your Algolia `${param:ALGOLIA_INDEX_NAME}` index.
 
-The `npm run import` is for use with the official Firebase Extension [**Search with Algolia**](https://github.com/algolia/algolia-firebase-extension).
+The `npm run import` is for use with the official Firebase Extension [**Search with Algolia**](https://github.com/algolia/firestore-algolia-search).
```
Comment (kevinthecheung) [inline POSTINSTALL.md:23]: I think you can omit this line, since the Algolia Extension context is already clear.

File: POSTINSTALL.md [hunk 2]
```diff
@@ -31,21 +31,33 @@ The `npm run import` is for use with the official Firebase Extension [**Search w
...
-- Algolia API Key: What is the Algolia API key with addObject, deleteObject, listIndexes, deleteIndex, editSettings, and settings permissions.
+- Algolia API Key: What is your Algolia API key?
+  We recommend creating a new [api key](https://www.algolia.com/doc/guides/security/api-keys/#creating-and-managing-api-keys)
+  with "addObject", "deleteObject", "listIndexes", "deleteIndex", "editSettings", and "settings" permissions.
+  **Do not use the Admin API key**.
 
-- Algolia Index Name: What is the Algolia index name that will contain the collection document records.
+- Algolia Index Name: What is the Algolia index name?
+  This is the name of the store where the records will be persisted.
+  Refer to [naming your index](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/#naming-your-index).
```
Comment (kevinthecheung) [inline POSTINSTALL.md:58]: IIUC, users will already have made this configuration (i.e., when they installed the extension) by the time they get to the point of running this script. If so, I don't think you need to list them here.

File: POSTINSTALL.md [hunk 3]
```diff
@@ -56,10 +68,11 @@ Run the import script using [`npx` (the Node Package Runner)](https://www.npmjs.
 
     Please follow the instructions to [generate a key for your service account](https://firebase.google.com/docs/admin/setup#initialize-sdk).
 
-1.  Clone [algolia-firebase-extension](https://github.com/algolia/algolia-firebase-extension) repo.
-1.  Run the import script via `npx`.
+1.  Clone [firestore-algolia-search](https://github.com/algolia/firestore-algolia-search) repo.
+1.  Execute the following command in the `functions` folder:
```
Comment (kevinthecheung) [inline POSTINSTALL.md:72]: 'commands'?

File: PREINSTALL.md [hunk 4]
```diff
@@ -17,16 +17,19 @@ extension. You can do so on the [Algolia](https://www.algolia.com/) site.
...
-    - Cloud Firestore
+- Cloud Firestore
+- Cloud Functions
 
-[blaze-pricing]: https://firebase.google.com/pricing
-[pricing-examples]: https://cloud.google.com/functions/pricing#pricing_examples
-[faq]: https://firebase.google.com/support/faq#expandable-24
+This extension also uses the following third-party services:
+
+- Algolia Billing ([pricing information](https://www.algolia.com/pricing))
```
Comment (kevinthecheung) [inline PREINSTALL.md:27]: Just "Algolia", not "Algolia Billing", right?

Comment (smomin) [reply PREINSTALL.md:27]: right.  i will correct it.

File: POSTINSTALL.md (hunk 3 above)
Comment (smomin) [reply POSTINSTALL.md:72]: there is only one command.

File: README.md [hunk 5]
```diff
@@ -26,10 +26,10 @@ firebase ext:install aloglia-firebase/firestore-algolia-search --project=<your-p
 
 Use this extension to index your Cloud Firestore collection.  The extension is applied and configured on a collection.
 
-This extension listens for changes to the specified collection. If a document is added, updated, or deleted, this
+This extension listens for changes on the specified collection. If a document is added, updated, or deleted, this
```
Comment (Meschreiber) [inline README.md:29]: Would recommend the same suggestions here as in the PREINSTALL guide, but only putting them in one place so as to not duplicate/maintain both sets of instructions. (Seems like the README might be a better place than PREINSTALL.)

File: README.md [hunk 6]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+- Cloud Functions Location: Where do you want to deploy the functions created for this extension?
+  You usually want a location close to your database.
+  For help selecting a location, refer to the
+  [location selection guide](https://firebase.google.com/docs/functions/locations).
 
-- Collection Path: What is the path to the collection that you want to index?
+- Collection Path: What is the path to the Cloud Firestore collection where the extension should monitor for changes?
 
-- Fields: What are fields in the document do you want to index?
+- Fields: What document fields should be indexed to provide the best search experience?
```
Comment (Meschreiber) [inline README.md:67]:
```suggestion
- Fields: What document fields should be indexed to provide the best search experience? For more information on which fields to index to Algolia, see the [Algolia documentation on records](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/#algolia-records).
```

File: README.md [hunk 7]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+  This can be comma separated list or left blank to index all fields.
+  Refer to [reducing record size](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/how-to/reducing-object-size/).
```
Comment (Meschreiber) [inline README.md:69]: I would put this after the record size limit suggestion.
```suggestion
```

File: POSTINSTALL.md (hunk 2 above)
Comment (smomin) [reply POSTINSTALL.md:58]: Can you provide more clarity here?

File: README.md [hunk 8]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+  For performance reasons, [record size is limited](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/in-depth/index-and-records-size-and-usage-limitations/#record-size-limits).
```
Comment (Meschreiber) [inline README.md:70]:
```suggestion
  For performance reasons, [record size is limited](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/in-depth/index-and-records-size-and-usage-limitations/#record-size-limits).
  If you're receiving errors that your records are too large,  refer to the [reducing record size documentation](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/how-to/reducing-object-size/).
```

File: README.md [hunk 9]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
 
-**NOTE:** This configuration can be left empty if you want to index all the fields in this document.
+- Algolia Application ID: What is the Algolia Application Id?
+  This is the application that contains one or more indexes.
```
Comment (Meschreiber) [inline README.md:73]:
```suggestion
  This is the Algolia application you want to index your data to.
```

File: README.md [hunk 10]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+  Refer to [application id](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/how-to/importing-with-the-api/#application-id).
```
Comment (Meschreiber) [inline README.md:74]:
```suggestion
You can find your credentials including application ID on your Algolia dashboard, under the [API keys tab](https://www.algolia.com/api-keys).
```

File: README.md [hunk 11]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
 
-- Algolia Application ID: What is the Algolia application id that contains the index?
+- Algolia API Key: What is your Algolia API key?
+  We recommend creating a new [api key](https://www.algolia.com/doc/guides/security/api-keys/#creating-and-managing-api-keys)
```
Comment (Meschreiber) [inline README.md:77]:
```suggestion
  We recommend [creating a new API key](https://www.algolia.com/doc/guides/security/api-keys/#creating-and-managing-api-keys)
```

File: README.md [hunk 12]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+  with "addObject", "deleteObject", "listIndexes", "deleteIndex", "editSettings", and "settings" permissions.
+  **Do not use the Admin API key**.
 
-- Algolia API Key: What is the Algolia API key with addObject, deleteObject, listIndexes, deleteIndex, editSettings, and settings permissions.
-
-- Algolia Index Name: What is the Algolia index name that will contain the collection document records.
+- Algolia Index Name: What is the Algolia index name?
+  This is the name of the store where the records will be persisted.
```
Comment (Meschreiber) [inline README.md:82]:
```suggestion
  This is the name of the [Algolia index](https://www.algolia.com/doc/faq/basics/what-is-an-index/) where the records will be persisted.
```

File: README.md [hunk 13]
```diff
@@ -57,21 +57,32 @@ To install an extension, your project must be on the
...
+  Refer to [naming your index](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/#naming-your-index).
```
Comment (Meschreiber) [inline README.md:83]:
```suggestion
  Refer to [naming your index](https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/#naming-your-index) for more information.
```

File: extension.yaml [hunk 14]
```diff
@@ -58,43 +58,53 @@ resources:
...
+    validationErrorMessage: Firestore collection paths must be an odd number of segments separated by slashes, e.g. "path/to/collection".
     required: true
 
   - param: FIELDS
     label: Indexable Fields
     description: >
-      A comma separated list of Cloud Firestore fields to index. Leave empty to index all fields.
+      What document fields should be indexed to provide the best search experience?
+      This can be comma separated list or left blank to index all fields.
+      Refer to [reducing record size](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/how-to/reducing-object-size/).
```
Comment (Meschreiber) [inline extension.yaml:75]:
```suggestion
       For more information on which fields to index to Algolia, see the [Algolia documentation on records](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/#algolia-records).
```

File: extension.yaml [hunk 15]
```diff
@@ -58,43 +58,53 @@ resources:
...
+      For performance reasons, [record size is limited](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/in-depth/index-and-records-size-and-usage-limitations/#record-size-limits).
```
Comment (Meschreiber) [inline extension.yaml:76]: Would recommend the other changes in the README applied to the description fields here as well.

**Review by Meschreiber — CHANGES_REQUESTED** [review]

File: PREINSTALL.md [hunk 16]
```diff
@@ -1,9 +1,9 @@
-Use this extension to index your Cloud Firestore collection for full text search with Algolia.  The extension is applied and configured on a collection.
+Use this extension to index your Cloud Firestore collection.  The extension is applied and configured on a collection.
 
-This extension listens for changes to the specified collection. If a document is added, updated, or deleted, this
+This extension listens for changes on the specified collection. If a document is added, updated, or deleted, this
 extension will:
 
-- Add/Update - document will be indexed into Algolia.  The fields defined in the extension configuration will be used to send to Algolia.  The document Id will be used as the object id in Algolia.
+- Add/Update - document will be indexed into Algolia.  The fields defined in the extension configuration will be used to capture the values and send to Algolia.  The collection document Id will be used as the Algolia record id.
```
[the remark this answers is not in the document]
Comment (smomin) [reply PREINSTALL.md:6]: Document ids are unique within the same collection.  It might be the same as the document name.

File: README.md (hunk 5 above)
Comment (smomin) [reply README.md:29]: I think we might need to deal with duplicates.  The readme is shown when you visit the Github repo.  The Preinstall is shown in the Extension landing page.

**Pull request merged**